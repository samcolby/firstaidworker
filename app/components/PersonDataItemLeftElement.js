import React, { memo } from "react";
import PropTypes from "prop-types";

import { StyleSheet, Text } from "react-native";
import { Icon } from "react-native-elements";

import { COLORS } from "../Constants";

function PersonDataItemLeftElement({ icon, iconType, title }) {
  return (
    <>
      <Icon
        name={icon}
        size={24}
        type={iconType}
        color={COLORS.NAV_HINTS}
        containerStyle={styles.iconContainer}
        iconStyle={styles.iconStyle}
      />
      <Text style={styles.titleStyle}>{title}</Text>
    </>
  );
}

PersonDataItemLeftElement.propTypes = {
  icon: PropTypes.string.isRequired,
  iconType: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired
};

var styles = StyleSheet.create({
  iconContainer: {
    justifyContent: "center",
    width: 18
  },
  iconStyle: { opacity: 0.9 },
  titleStyle: {
    color: COLORS.NAV_HINTS,
    opacity: 0.9,
    fontSize: 14,
    marginLeft: 6,
    width: 64
  }
});

export default memo(PersonDataItemLeftElement);
