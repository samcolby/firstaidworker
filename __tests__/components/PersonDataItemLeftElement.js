import React from "react";
import { Text } from "react-native";
import { Icon } from "react-native-elements";

import { shallow } from "enzyme";

import PersonDataItemLeftElement from "../../app/components/PersonDataItemLeftElement";

describe("PersonDataItemLeftElement Component", () => {
  it("should render without issues", () => {
    const component = shallow(
      <PersonDataItemLeftElement
        icon="ios-business"
        iconType="ionicons"
        title="This is the title"
        value=""
      />
    );

    expect(component.length).toBe(1);
    expect(component).toMatchSnapshot();
  });

  it("displays the title", () => {
    const component = shallow(
      <PersonDataItemLeftElement
        icon="ios-business"
        iconType="ionicons"
        title="This is the title"
        value=""
      />
    );

    expect(
      component.containsMatchingElement(<Text>This is the title</Text>)
    ).toBe(true);
  });

  it("displays the correct iconType", () => {
    const component = shallow(
      <PersonDataItemLeftElement
        icon="ios-business"
        iconType="ionicons"
        title="This is the title"
        value=""
      />
    );

    const iconComponent = component.find(Icon);
    expect(iconComponent.prop("type")).toEqual("ionicons");
  });

  it("displays the correct icon", () => {
    const component = shallow(
      <PersonDataItemLeftElement
        icon="ios-business"
        iconType="ionicons"
        title="This is the title"
        value=""
      />
    );

    const iconComponent = component.find(Icon);
    expect(iconComponent.prop("name")).toEqual("ios-business");
  });
});
