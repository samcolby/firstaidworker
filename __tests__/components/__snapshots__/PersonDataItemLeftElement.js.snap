// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonDataItemLeftElement Component should render without issues 1`] = `
<Fragment>
  <Themed.Icon
    color="#ff5454"
    containerStyle={
      Object {
        "justifyContent": "center",
        "width": 18,
      }
    }
    iconStyle={
      Object {
        "opacity": 0.9,
      }
    }
    name="ios-business"
    size={24}
    type="ionicons"
  />
  <Text
    style={
      Object {
        "color": "#ff5454",
        "fontSize": 14,
        "marginLeft": 6,
        "opacity": 0.9,
        "width": 64,
      }
    }
  >
    This is the title
  </Text>
</Fragment>
`;
